addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -6049,14 +6057,28 @@ registry.BackgroundImage = SnippetOptionWidget.extend({
      */
     _setBackground(backgroundURL) {
         const parts = backgroundImageCssToParts(this.$target.css('background-image'));
+        let wasDefaultGradient = false;
         if (backgroundURL) {
             parts.url = `url('${backgroundURL}')`;
             this.$target.addClass('oe_img_bg o_bg_img_center');
         } else {
             delete parts.url;
             this.$target.removeClass('oe_img_bg o_bg_img_center');
+            if (parts.gradient) {
+                // If gradient is the default one then remove it.
+                this.$target.css('background-image', '');
+                const defaultParts = backgroundImageCssToParts(this.$target.css('background-image'));
+                if (parts.gradient && defaultParts.gradient && weUtils.areCssValuesEqual(parts.gradient, defaultParts.gradient)) {
+                    delete parts.gradient;
+                    wasDefaultGradient = true;
+                }
+            }
+        }
+        let combined = backgroundImagePartsToCss(parts);
+        if (combined === 'none' && wasDefaultGradient) {
+            // Do not hide default gradient.
+            combined = '';
```

Comment:
Mmh, I guess it's ok if there is no other way but I feel like there is something better to do here: normally our `selectStyle` method is smart enough to not apply a given style as inline style if it would have no effect (because already set by a CSS rule). I think this could reuse the same code here, either by directly calling `selectStyle` or an extracted part of it instead of the final call here `this.$target.css('background-image', combined);`. So the part of your code with `// If gradient is the default one then remove it.` would be automatically done by trying to set the same gradient as the CSS rule as background image. Could you check if I am right? :thinking:

Reply:
It actually works - even when `combined` is `"none"`. This is much simpler indeed.